orchagent/mplsrouteorch.cpp

```diff
@@ -251,20 +291,24 @@ void RouteOrch::doLabelTask(Consumer& consumer)
                 }
                 else if (m_syncdLabelRoutes.find(vrf_id) == m_syncdLabelRoutes.end() ||
                          m_syncdLabelRoutes.at(vrf_id).find(label) == m_syncdLabelRoutes.at(vrf_id).end() ||
-                         m_syncdLabelRoutes.at(vrf_id).at(label) != nhg)
+                         m_syncdLabelRoutes.at(vrf_id).at(label) != RouteNhg(nhg, nhg_index) ||
+                         ctx.using_temp_nhg)
                 {
                     if (addLabelRoute(ctx, nhg))
                         it = consumer.m_toSync.erase(it);
                     else
                         it++;
                 }
                 else
+                {
                     /* Duplicate entry */
+                    SWSS_LOG_INFO("Route %s is duplicate entry", key.c_str());
                     it = consumer.m_toSync.erase(it);
+                }
 
                 // If already exhaust the nexthop groups, and there are pending removing routes in bulker,
                 // flush the bulker and possibly collect some released nexthop groups
-                if (m_nextHopGroupCount >= m_maxNextHopGroupCount &&
+                if (gNhgOrch->getNhgCount() >= gNhgOrch->getMaxNhgCount() &&
```

Comment:
what happens if the routeOrch has consumed nexthop resources?

Reply:
nhgOrch owns the count of next hop groups - routeOrch and mplsRouteOrch both call into nhgOrch to increment/decrement its count. So gNhgOrch->getNhgCount() is getting the total count of all next hop groups on the system